msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -3,58 +3,84 @@
     }
 
     /**
      * Acquire a token silently, i.e. without direct user interaction, using username/password authentication
-     *
-     * @param requestParameters MsalRequest object which contains everything needed for the broker implementation to make a request
-     * @return IBroker implementations will return an AuthenticationResult object
      */
-    default IAuthenticationResult acquireToken(PublicClientApplication application, UserNamePasswordParameters requestParameters) {
+    default void acquireToken(PublicClientApplication application, UserNamePasswordParameters parameters, CompletableFuture<IAuthenticationResult> future) {
         throw new MsalClientException("Broker implementation missing", AuthenticationErrorCode.MISSING_BROKER);
     }
 
-    default CompletableFuture removeAccount(IAccount account) {
+    default void removeAccount(PublicClientApplication application, IAccount account) throws MsalClientException {
         throw new MsalClientException("Broker implementation missing", AuthenticationErrorCode.MISSING_BROKER);
     }
+
+    //TODO: Any better place to put this helper method? This feels wrong
+    //AuthenticationResult requires many package-private classes that broker package can't access, so helper methods will be
+    //  made for each broker to create the sort of AuthenticationResult that the rest of MSAL Java expects
+    default IAuthenticationResult parseBrokerAuthResult(String authority, String idToken, String accessToken,
+                                                            String accountId, String clientInfo,
+                                                            long accessTokenExpirationTime) {
+
+        //TODO: need to either make AuthenticationResult public or implement IAuthenticationResult here in the interop layer
+        AuthenticationResult.AuthenticationResultBuilder builder =  AuthenticationResult.builder();
+
+        try {
+            if (idToken != null) {
+                builder.idToken(idToken);
+                if (accountId!= null) {
+                    String idTokenJson =
+                            JWTParser.parse(idToken).getParsedParts()[1].decodeToString();
+                    //TODO: need to figure out if 'policy' field is relevant for brokers
+                    builder.accountCacheEntity(AccountCacheEntity.create(clientInfo,
+                            Authority.createAuthority(new URL(authority)), JsonHelper.convertJsonToObject(idTokenJson,
+                                    IdToken.class), null));
+                }
+            }
+            if (accessToken != null) {
+                builder.accessToken(accessToken);
+                builder.expiresOn(accessTokenExpirationTime);
+            }
+        } catch (Exception e) {
+            //TODO: throw new msalexception. Could a valid broker result be an invalid MSAL Java result?
```

Comment:
Ideally, the code should not reach here, but an exception could happen while converting json to object, parsing an id token, etc - in which case it would come under an invalid broker result. But it is good to throw a meaningful MSALException here, and we can figure out if we should pass this error back to MSALRuntime somehow to let them know about the invalid broker results.